js_modules/dagster-ui/packages/ui-components/src/theme/GlobalThemeStyle.tsx

```diff
@@ -0,0 +1,40 @@
+import {createGlobalStyle} from 'styled-components';
+
+import {darkThemeColors} from './darkThemeColors';
+import {lightThemeColors} from './lightThemeColors';
+import {CoreColorStyles} from '../palettes/CoreColorStyles';
+import {DataVizColorStyles} from '../palettes/DataVizColorStyles';
+import {TranslucentColorStyles} from '../palettes/TranslucentColorStyles';
+
+const ThemeRoot = createGlobalStyle`
+  @media (prefers-color-scheme: light) {
+    :root, .themeSystem {
+      ${lightThemeColors}
+    }
+  }
+
+  @media (prefers-color-scheme: dark) {
+    :root, .themeSystem {
+      ${darkThemeColors}
+    }
+  }
+
+  .themeLight {
+    ${lightThemeColors}
+  }
+
+  .themeDark {
+    ${darkThemeColors}
+  }
+`;
```

Comment:
A shame this will end up a giant string including the same vars twice.
Style queries could help here https://developer.chrome.com/docs/css-ui/style-queries but the support is limited.